worlds/sc2/mission_order/options.py

```diff
@@ -0,0 +1,381 @@
+from .structs import Difficulty, LayoutType
+from .types import Column, Grid, Hopscotch, Gauntlet, Blitz
+from .presets_static import (
+    static_preset, preset_mini_wol_with_prophecy, preset_mini_wol, preset_mini_hots, preset_mini_prophecy,
+    preset_mini_lotv_prologue, preset_mini_lotv, preset_mini_lotv_epilogue, preset_mini_nco,
+    preset_wol_with_prophecy, preset_wol, preset_prophecy, preset_hots, preset_lotv_prologue,
+    preset_lotv_epilogue, preset_lotv, preset_nco
+)
+from .presets_scripted import make_golden_path
+
+STR_OPTION_VALUES = {
+    "type": {
+        "column": Column, "grid": Grid, "hopscotch": Hopscotch, "gauntlet": Gauntlet, "blitz": Blitz,
+    },
+    "difficulty": {
+        "relative": Difficulty.RELATIVE, "starter": Difficulty.STARTER, "easy": Difficulty.EASY,
+        "medium": Difficulty.MEDIUM, "hard": Difficulty.HARD, "very hard": Difficulty.VERY_HARD
+    },
+    "preset": {
+        "none": lambda _: {},
+        "wol + prophecy": static_preset(preset_wol_with_prophecy),
+        "wol": static_preset(preset_wol),
+        "prophecy": static_preset(preset_prophecy),
+        "hots": static_preset(preset_hots),
+        "prologue": static_preset(preset_lotv_prologue),
+        "lotv prologue": static_preset(preset_lotv_prologue),
+        "lotv": static_preset(preset_lotv),
+        "epilogue": static_preset(preset_lotv_epilogue),
+        "lotv epilogue": static_preset(preset_lotv_epilogue),
+        "nco": static_preset(preset_nco),
+        "mini wol + prophecy": static_preset(preset_mini_wol_with_prophecy),
+        "mini wol": static_preset(preset_mini_wol),
+        "mini prophecy": static_preset(preset_mini_prophecy),
+        "mini hots": static_preset(preset_mini_hots),
+        "mini prologue": static_preset(preset_mini_lotv_prologue),
+        "mini lotv prologue": static_preset(preset_mini_lotv_prologue),
+        "mini lotv": static_preset(preset_mini_lotv),
+        "mini epilogue": static_preset(preset_mini_lotv_epilogue),
+        "mini lotv epilogue": static_preset(preset_mini_lotv_epilogue),
+        "mini nco": static_preset(preset_mini_nco),
+        "golden path": make_golden_path
+    },
+}
+STR_OPTION_VALUES["min_difficulty"] = STR_OPTION_VALUES["difficulty"]
+STR_OPTION_VALUES["max_difficulty"] = STR_OPTION_VALUES["difficulty"]
+GLOBAL_ENTRY = "global"
+
+StrOption = lambda cat: And(str, lambda val: val.lower() in STR_OPTION_VALUES[cat])
+IntNegOne = And(int, lambda val: val >= -1)
+IntZero = And(int, lambda val: val >= 0)
+IntOne = And(int, lambda val: val >= 1)
+IntPercent = And(int, lambda val: 0 <= val <= 100)
+
+SubRuleEntryRule = {
+    "rules": [{str: object}], # recursive schema checking is too hard
+    "amount": IntNegOne,
+}
+MissionCountEntryRule = {
+    "scope": [str],
+    "amount": IntNegOne,
+}
+BeatMissionsEntryRule = {
+    "scope": [str],
+}
+EntryRule = Or(SubRuleEntryRule, MissionCountEntryRule, BeatMissionsEntryRule)
+
+class CustomMissionOrder(OptionDict):
```

Comment:
I think Berserker added a visibility flag field that we can use here to make this invisible in the GUI options page. I don't think this will function at all in the GUI page, but it may be worth testing with webhost.py before committing.

I see the default set on `class Option` in Options.py, line 120: `visibility = Visiblity.all`. We'd probably want to turn it off for `simple_ui` and `complex_ui`, and maybe check how parsable this is in the spoiler before deciding whether to leave it there.